examples/decomposition/plot_kernel_pca.py

```diff
@@ -3,78 +3,245 @@
+# original data was already maximized. However, we can see that the data have
+# been rotated. As a conclusion, we see that such a projection would not help
+# if define a linear classifier to distinguish samples from both classes.
+#
+# Using a kernel allows to make a non-linear projection. Here, by using an RBF
+# kernel, we expect that the projection to unfold the dataset but keeping that
+# point close in the original space should still be close in the new space.
+#
+# We observe such behaviour in the bottom-right figure: the samples of a given
+# class are closer to each other than the samples from the opposite class. The
+# "radial" effect make that we unrolled the circle. Now, we can use a linear
+# classifier to separate the samples from the two classes.
+#
+# Projecting into the original feature space
+# ------------------------------------------
+#
+# One particularity to have in mind when using
+# :class:`~sklearn.decomposition.KernelPCA` is related to the reconstruction
+# (i.e. the back projection in the original feature space). With
+# :class:`~sklearn.decomposition.PCA`, the reconstruction will be exact if
+# `n_components` is the same than the number of original features as in this
+# example. Thus, projecting the data on the PCA basis and projecting back will
+# give the same dataset.
+#
+# We can investigate if we get a similar outcome with
+# :class:`~sklearn.decomposition.KernelPCA`.
+X_reconstructed_pca = pca.inverse_transform(pca.transform(X_test))
+X_reconstructed_kernel_pca = kernel_pca.inverse_transform(
+    kernel_pca.transform(X_test))
+
+# %%
+fig, axs = plt.subplots(ncols=3, sharex=True, sharey=True, figsize=(13, 4))
+
+axs[0].scatter(X_test[:, 0], X_test[:, 1], c=y_test)
+axs[0].set_ylabel("Feature #1")
+axs[0].set_xlabel("Feature #0")
+axs[0].set_title("Original test data")
+
+axs[1].scatter(X_reconstructed_pca[:, 0], X_reconstructed_pca[:, 1], c=y_test)
+axs[1].set_xlabel("Feature #0")
+axs[1].set_title("Reconstruction via PCA")
+
+axs[2].scatter(X_reconstructed_kernel_pca[:, 0],
+               X_reconstructed_kernel_pca[:, 1], c=y_test)
+axs[2].set_xlabel("Feature #0")
+_ = axs[2].set_title("Reconstruction via KernelPCA")
+
+# %%
+# While we see a perfect reconstruction, we observe a different results for
+# :class:`~sklearn.decomposition.KernelPCA`. Indeed,
+# :meth:`~sklearn.decomposition.KernelPCA.inverse_transform` cannot rely on an
+# analytical back-projection and thus an extact reconstruction. Instead, a
+# :class:`~sklearn.linear_model.KernelRidge` was trained to learn a projection
+# function to map a sample from the PCA basis into the original feature space.
+# This method is therefore an approximation leading to small difference. The
+# parameter `alpha` in the :class:`~sklearn.decomposition.KernelPCA` is used
+# to penalized the mapping function to fit more or less the training data.
+#
+# Application to image denoising
+# ------------------------------
```

Comment:
This feels like a new example deserving its own file / page.

Reply:
OK, this could make sense